lib/rpc/GlobalRpcProviders.ts

```diff
@@ -0,0 +1,69 @@
+import { ChainId } from '@uniswap/sdk-core'
+import { SingleJsonRpcProvider } from './SingleJsonRpcProvider'
+import { UniJsonRpcProvider } from './UniJsonRpcProvider'
+import Logger from 'bunyan'
+
+export class GlobalRpcProviders {
+  private static readonly PROVIDER_RPC_URL_RANKING: Map<ChainId, number[] | undefined> = new Map([
+    [ChainId.MAINNET, undefined],
+  ])
+
+  private static readonly PROVIDER_RPC_URL_WEIGHTS: Map<ChainId, number[] | undefined> = new Map([
+    [ChainId.MAINNET, undefined],
+  ])
+
+  private static SINGLE_RPC_PROVIDERS: Map<ChainId, SingleJsonRpcProvider[]> | null = null
+
+  private static UNI_RPC_PROVIDERS: Map<ChainId, UniJsonRpcProvider> | null = null
+
+  private static initGlobalSingleRpcProviders(log: Logger) {
+    const INFURA_KEY = process.env.UNI_RPC_PROVIDER_INFURA_KEY
+    if (INFURA_KEY === undefined) {
+      throw new Error(`UNI_RPC_PROVIDER_INFURA_KEY must be a defined environment variable`)
+    }
+    const QUICKNODE_MAINNET_RPC_URL = process.env.UNI_RPC_PROVIDER_QUICKNODE_MAINNET_RPC_URL
+    if (QUICKNODE_MAINNET_RPC_URL === undefined) {
+      throw new Error(`UNI_RPC_PROVIDER_QUICKNODE_MAINNET_RPC_URL must be a defined environment variable`)
+    }
+    this.SINGLE_RPC_PROVIDERS = new Map([
+      [
+        ChainId.MAINNET,
+        [
+          new SingleJsonRpcProvider(ChainId.MAINNET, `https://mainnet.infura.io/v3/${INFURA_KEY}`, log),
+          new SingleJsonRpcProvider(ChainId.MAINNET, QUICKNODE_MAINNET_RPC_URL!, log),
+        ],
+      ],
+    ])
+    return this.SINGLE_RPC_PROVIDERS
+  }
+
+  private static initGlobalUniRpcProviders(log: Logger) {
+    if (this.SINGLE_RPC_PROVIDERS === null) {
+      this.initGlobalSingleRpcProviders(log)
+    }
+    if (!this.SINGLE_RPC_PROVIDERS!.has(ChainId.MAINNET)) {
+      throw new Error(`No RPC providers configured for chain ${ChainId.MAINNET}`)
+    }
+    this.UNI_RPC_PROVIDERS = new Map([
+      [
+        ChainId.MAINNET,
+        new UniJsonRpcProvider(
+          ChainId.MAINNET,
+          this.SINGLE_RPC_PROVIDERS!.get(ChainId.MAINNET)!,
+          log,
+          GlobalRpcProviders.PROVIDER_RPC_URL_RANKING.get(ChainId.MAINNET),
```

Comment:
don't forget to change to a non-mainnet before merging this PR. We don't have feature switch for provider gateway fallback, so it will take effect in prod immediately, upon merging the PR. Because by merging the PR, it will automatically deploy to staging and then prod (assuming the pipeline releases successfully)

Reply:
no code is actually reading from it so it should not matter right?